### ISSUE TITLE
UPE visual placeholder while mounting

### ISSUE THREAD

**Issue by bborman22** [Automattic/woocommerce-payments#1987]
There is a time delay when loading the UPE. This causes the payment area to be blank while the payment intent is created and the UPE is mounted.

https://d.pr/i/UGfvuR

This delay is due to the need to create a payment intent before loading in the UPE. There was some initial discussion on creating the intent before loading the checkout page, but due to another conflicting issue we decided to opt towards creating it after page load. This also helps the checkout page to load faster initially, but causes this longer delay in mounting the UPE.

The goal of this issue is to make sure that there is a good visual experience while loading the UPE. One example would be to add a fake "placeholder" element while the real UPE is loading similar to what the current WCPay gateway does.

**Comment by bborman22** [comment]
@jarekmorawski This will be something coming down the line in a little bit and wanted to ping you for any additional input. There is nothing urgent on this issue for right now, but wanted to give plenty of runway for any input.

**Comment by jarekmorawski** [comment]
Hi @bborman22!

Thanks for bringing this to my attention. Do you think we could add a skeleton loading screen inside the UPE container?

https://d.pr/i/al6tZh

**Comment by bborman22** [comment]
So I had experimented with a few things and with the different variables (only one payment method, card vs giropay, etc) and found that trying to match up the skeleton fields made it a bit more confusing. For example cards have the grid with 5 inputs and Giropay only has one input with text below it. I opted for a simpler solution of just dropping a loading icon in place before it mounts.

@jarekmorawski I was wondering what you thought of this: https://d.pr/i/F2jdhj
